pkg/node/connection.go

```diff
@@ -253,6 +254,21 @@ func (e *Node) handleEvents(ctx context.Context, inputChannel chan *hub.Message,
 				continue
 			}
 
+			// If we have enabled trusted arbiter peers
+			if len(e.config.TrustedPeerIDS) > 0 && e.host.ID().String() != m.SenderID {
```

Comment:
it would make sense to be consistent here and instantiate the PeerGater with static trusted peer IDs.

For instance, the gater constructor could take as arguments a set of trustedPeerIDs coming from the config

https://github.com/mudler/edgevpn/blob/a11f392658ea701c9a5a6e01cc30f17cd2a1de0c/pkg/trustzone/peergater.go#L35

and `Gate()` could check both in `trustedDB` (coming from the blockchain) and the static ones (that you are adding) coming from the constructor

Reply:
i have reviewed that now more deeply, as i have returned to edgevpn code while doing LocalAI PR init commit - the thing is, the configured initial trust IDs should work as a switch for authoritative/passive mode, where we either gating messages based on trustDB in authoritative mode, or gating all messages that came not from the initial trusted IDs in passive mode

The peergater most likely remains the same, and stays in the scope of working only for blockchain trustDB

Reply:
right, but this sounds like implementing a whitelisting mechanism which is very much similar to the PeerGater. To keep code concise and clean we could or either expand the PeerGater module to accept a static ids and act it as a whitelister, or have a separate component dedicated to whitelisting (which would be even more clean).

Reply:
i think i will expand peergater then, as for now whitelisting is effective only in pair with peergater

Reply:
yet, the ProtectedStoreKeys is relying at TrustedPeerIDs, so most likely the separated whitelisting component is needed